### PR TITLE
Whisper pipeline: support long-form audio

### PR DESCRIPTION
**Pull request by as-suvorov**
This PR adds:
- [x] Long-form audio support with sequential chunking.

Common Todos for Whisper support:
- [ ] Long-form audio support with [parallel chunking](https://huggingface.co/blog/asr-chunking).
- [ ] add perf metrics
- [ ] update documentation
- [ ] add cpp, python samples tests
- [ ] support timestamps streaming
- [ ] expose only meaningful parameters in `GenerationConfig` (`task`, `language`, `return_timestamps`, etc)
- [ ] Move all whisper pipeline files to dedicated subfolder
- [ ] Whisper pipeline doesn't need tokenizer, it uses detokenizer only. Implement detokenizer only initialization for `ov::genai::Tokenizer`
- [ ] Check discrete GPU. Integrated GPU works as expected.
- [ ] Investigate use of `RemoteTensor` for GPU
- [ ] Add batch
- [ ] Add sampler, inherit WhisperGenerationConfig from GenerationConfig
- [ ] Investigate language autodetection with single decoder (without past) call
- [ ] Update python bindings cmake to include whole directory instead of explicit list of files
- [ ] Add samples with audio preparation examples
- [ ] Add links to audio files so users can download them in samples
- [ ] Move supported models list from samples README to common supported models section
- [ ] Avoid building GenAI in each tests job as it takes a lot of time
- [ ] Double check FP32 support
- [ ] Fix tests sporadic fails. Sometimes whisper model cannot be downloaded from HF due to network issues
- [ ] Fix stop criteria. Current approach stops on eos_token which is no speech token. But there could be more speech tokens further which are wrongly skipped now.

Completed:
- [x] support different languages, language autodetection
- [x] support translation
- [x] support timestamps

Current limitations:
- No resampling during preprocessing. Input raw speech should have 16k Hz sampling rate
- No normalization during preprocessing. Input raw speech should be normalized to near [-1, 1] range

Tickets: CVS-147994, CVS-146010, CVS-152542

### CODE REVIEW

**Comment by as-suvorov** [comment]
Depends on: https://github.com/openvinotoolkit/openvino.genai/pull/910

File: samples/python/whisper_speech_recognition/whisper_speech_recognition.py [hunk 1]
```diff
@@ -41,6 +41,11 @@ def streamer(word: str) -> bool:
 
     print()
 
+    for chunk in result.chunks:
+        print(f"timestamps: [{chunk.start_ts}, {chunk.end_ts}] text: {chunk.text}")
+
+    print()
```
Comment (ilya-lavrenov) [inline samples/python/whisper_speech_recognition/whisper_speech_recognition.py:47]: duplication because of merge conflicts?

File: samples/generation.gif
Comment (ilya-lavrenov) [inline samples/generation.gif:1]: please, keep these image files unchanged

**Review by ilya-lavrenov — APPROVED** [review]

File: samples/python/whisper_speech_recognition/whisper_speech_recognition.py (hunk 1 above)
Comment (as-suvorov) [reply samples/python/whisper_speech_recognition/whisper_speech_recognition.py]: Yes, thanks for the catch!

File: samples/generation.gif
Comment (as-suvorov) [reply samples/generation.gif:1]: Fixed

**Review by sbalandi — APPROVED** [review]

**Pull request merged**